client/azure.go

```diff
@@ -56,8 +58,8 @@ func DefaultAzureConfig() *AzureConfig {
 type AzureClientSet struct {
 	// DeploymentsClient manages deployments of ARM templates.
 	DeploymentsClient *resources.DeploymentsClient
-	// GroupClient manages ARM resource groups.
-	GroupClient *resources.GroupClient
+	// GroupsClient manages ARM resource groups.
+	GroupsClient *resources.GroupsClient
```

Comment:
I was using the wrong client here! The GroupClient is more generic and harder to work with. Whereas the GroupsClient is specifically for managing resource groups.